app/models/report.rb

```diff
@@ -46,4 +46,11 @@ def end_date
       start_date + duration.days
     end
   end
+
+  private
+
+  def active_components(opts={})
+    result = text_components.includes(:channels)
+    text_components.select {|c| c.active?(opts) }
+  end
```

Comment:
This isn’t directly related to the issues solved in this PR, but to #311. If I’m right, sensorstories as well as the chatbot API should both only include published text components. So it might be a good idea to filter for published text components in the `active_components` method in order not repeat ourselfs. The `visible_sensor_story_components` would then be superfluous, but the code would be way cleaner imo. (Yep, I know I have introduced that new `visible_*` method…)

If there are any plans for the future (?!) for a preview features that includes unpublished text components as well, it might still be a good idea to have a single method, that optionally returns unpublished components as well.

In any case: The chatfuel API should return published components only as well, right? Would be great to hear your thoughts on this, I’ll reopen #311 then and work this out.